compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -357,9 +357,16 @@ class Definitions {
     def newGenericArrayMethod(implicit ctx: Context) = DottyArraysModule.requiredMethod("newGenericArray")
     def newArrayMethod(implicit ctx: Context) = DottyArraysModule.requiredMethod("newArray")
 
+  lazy val ListType = ctx.requiredClassRef("scala.collection.immutable.List")
+  def ListClass = ListType.symbol.asClass
```

Comment:
This looks wrong. Don't we need to pass a context to `ListClass`, like we do everywhere else? (same for the other new definitions)

Reply:
Indeed, fixed.